source/tests/system/nidaqmx_driver_api_tests.cpp

```diff
@@ -705,7 +705,7 @@ class NiDAQmxDriverApiTests : public Test {
   template <typename TRequest>
   void set_request_session_id(TRequest& request, const nidevice_grpc::Session& session)
   {
-    request.mutable_task()->set_id(session.id());
+    request.mutable_task()->set_name(session.name());
```

Comment:
I agree.  I was somewhat disappointed to see how much code decomposes the session instead of passing it around.  In my world the session was an opaque structure for most code.